posthog/hogql/metadata.py

```diff
@@ -116,8 +126,15 @@ def process_expr_on_table(
 
 
 def is_valid_view(select_query: ast.SelectQuery | ast.SelectSetQuery) -> bool:
+    """Is not a valid view if:
+    a) There are any function calls in the select clause
+    b) There are any wildcard fields in the select clause
+    """
     for query in extract_select_queries(select_query):
         for field in query.select:
-            if not isinstance(field, ast.Alias):
+            if isinstance(field, ast.Call):
                 return False
+            if isinstance(field, ast.Field):
+                if field.chain and field.chain[-1] == "*":
+                    return False
```

Comment:
I'm going based off of the issue that was originally created. I think the reasoning is that function names might not necessarily serialise to a sensible field name and then we may run into issues in certain places (and by extension, `*` could also expand to include a function). I'm not too familiar with this but for now I just want to relax these constraints as much as possible and then look into removing them entirely in the future (but will need to spend a bit more time ensuring it will work properly)

Reply:
Cool, gotcha - yeah we currently serialize them into sensible names in the `columns` field of a HogQL query, so I imagine we should be able to reuse that logic for this (maybe)

Reply:
ah good to know, will keep that in mind!